nemo/collections/tts/modules/fastpitch.py

```diff
@@ -261,12 +276,37 @@ def forward(
             if self.learn_alignment and pitch.shape[-1] != pitch_predicted.shape[-1]:
                 # Pitch during training is per spectrogram frame, but during inference, it should be per character
                 pitch = average_pitch(pitch.unsqueeze(1), attn_hard_dur).squeeze(1)
+            elif not self.learn_alignment:
+                # If alignment is not learnt attn_hard_dur is None, hence durs_predicted is used in average_pitch
+                pitch = average_pitch(pitch.unsqueeze(1), durs_predicted).squeeze(1)
```

Comment:
The beginning of the forward method has:

```
if not self.learn_alignment and self.training:
  assert durs is not None
  assert pitch is not None
```
This implies that if alignment is not being learned, then the ground truth duration information will be passed into forward(). This ground truth "durs" is then what would have to be used for all of the training inference. Using the predicted durs would make sense if self.training is not set, though they can use the infer() function for that.

Trying to reverse engineer the intent for the old alignment code and support it without tests is more trouble than it is worth.

Reply:
I am not sure. As I mentioned before, I think we should have a separate dedicated PR for dealing with `self.learn_alignment`

Reply:
I am OK with leaving it to another PR. If we do that, should we revert the pitch change to be safe?